src/st/np3/player_water_effect.c

```diff
@@ -0,0 +1,343 @@
+                        if (var_s3 >= sp4A) {
+                            if (gPlayerUnk0C & 1) {
+                                var_s1 = var_s1 * 3 / 4;
+                            } else {
+                                var_s1 /= 2;
+                            }
+                        } else {
+                            var_s1 = (var_s1 * var_s3) / 80;
+                        }
+                        if (((var_s1 < 0) &&
+                             ((g_Player.pl_vram_flag & 8) == 0)) ||
+                            ((var_s1 > 0) &&
+                             ((g_Player.pl_vram_flag & 4) == 0))) {
+                            player->posX.val += var_s1 << 4;
+                        }
+                    }
+                }
+            }
+        }
+        if (arg0->ext.aluwater.unk7E) {
+            arg0->ext.aluwater.unk7E--;
+        }
+        *D_80097448 = var_s3;
+        if (gPlayerUnk0C & 0x27) {
+            if (gPlayerUnk0C & 0x20) {
+                if (gPlayerUnk0C & 4) {
+                    if (var_s3 > 4) {
+                        D_80097448[1] = var_s3 - 4;
+                    } else {
+                        D_80097448[1] = 0;
+                    }
+                    if (var_s3 > 12) {
+                        D_80097450 = var_s3 - 12;
+                    } else {
+                        D_80097450 = 0;
+                    }
+                } else {
+                    if (var_s3 > 8) {
+                        D_80097448[1] = var_s3 - 8;
+                    } else {
+                        D_80097448[1] = 0;
+                    }
+                    if (var_s3 > 0x18) {
+                        D_80097450 = var_s3 - 0x18;
+                    } else {
+                        D_80097450 = 0;
+                    }
+                }
+            } else if (gPlayerUnk0C & 3) {
+                if (var_s3 > 6) {
+                    D_80097448[1] = var_s3 - 6;
+                } else {
+                    D_80097448[1] = 0;
+                }
+                if (var_s3 > 0x10) {
+                    D_80097450 = var_s3 - 0x10;
+                } else {
+                    D_80097450 = 0;
+                }
+            } else if (gPlayerUnk0C & 4) {
+                if (var_s3 > 8) {
+                    D_80097448[1] = var_s3 - 8;
+                } else {
+                    D_80097448[1] = 0;
+                }
+                if (var_s3 > 0x18) {
+                    D_80097450 = var_s3 - 0x18;
+                } else {
+                    D_80097450 = 0;
+                }
+            }
+        } else {
+            if (var_s3 > 0x10) {
+                D_80097448[1] = var_s3 - 0x10;
+            } else {
+                D_80097448[1] = 0;
+            }
+            if (var_s3 > 0x30) {
+                D_80097450 = var_s3 - 0x30;
+            } else {
+                D_80097450 = 0;
+            }
+        }
+    } else {
+        InitializeEntity(&D_80180A60);
```

Comment:
```suggestion
        InitializeEntity(D_80180A60);
```
`D_80180A60` is an array

Reply:
Good catch, fixed